contracts/context-config/src/mutate.rs

```diff
@@ -306,20 +307,49 @@ impl ContextConfigs {
         context_id: Repr<ContextId>,
         account_id: AccountId,
     ) -> Promise {
-        // Deploy and initialize the proxy contract
-        Promise::new(account_id.clone())
+        // Known constants from NEAR protocol
+        //Use near tokens instead of yoctoNEAR
+        const ACCOUNT_CREATION_COST: u128 = 1_000_000_000_000_000_000_000; // 0.001 NEAR
+        const MIN_ACCOUNT_BALANCE: u128 = 3_500_000_000_000_000_000_000; // 0.0035 NEAR
+
+        // Calculate storage needs
+        let contract_bytes = self.proxy_code.get().clone().unwrap();
+        let storage_cost = (contract_bytes.len() as u128) * env::storage_byte_cost().as_yoctonear();
+
+        // Calculate required deposit
+        let required_deposit = NearToken::from_yoctonear(
+            ACCOUNT_CREATION_COST +    // Cost to create account
+            MIN_ACCOUNT_BALANCE +      // Minimum balance required
+            storage_cost +             // Storage cost for contract
+            10_000_000_000_000_000_000, // Additional deposit for storage
+        );
+
+        require!(
+            env::account_balance() >= required_deposit,
+            "Insufficient contract balance for deployment"
+        );
+
+        // Calculate init gas dynamically
+        let init_args = serde_json::to_vec(&json!({
+            "context_id": context_id,
+            "context_config_account_id": env::current_account_id()
+        }))
+        .unwrap();
+
+        let init_gas = Gas::from_gas(
+            Gas::from_tgas(20).as_gas()
+                + ((init_args.len() as u64) + contract_bytes.len() as u64) * 100_000,
```

Comment:
will `contract_bytes` be necessary here? since `init_gas` is only useful for the function call (which is already after deployment)